sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapi/AzureActiveDirectoryResourceServerClientConfigurationTest.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.aad.webapi;
+
+import com.azure.spring.aad.webapp.AzureClientRegistrationRepository;
+import org.junit.Test;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AzureActiveDirectoryResourceServerClientConfigurationTest {
+
+    private static final String AAD_PROPERTY_PREFIX = "azure.activedirectory.";
+
+    private final WebApplicationContextRunner contextRunner = new WebApplicationContextRunner()
+        .withPropertyValues(AAD_PROPERTY_PREFIX + "user-group.allowed-groups=group1",
+            AAD_PROPERTY_PREFIX + "tenant-id=fake-tenant-id",
+            AAD_PROPERTY_PREFIX + "client-id=fake-client-id",
+            AAD_PROPERTY_PREFIX + "client-secret=fake-client-secret",
+            AAD_PROPERTY_PREFIX + "web-application.enabled=false");
+
+    @Test
+    public void testNotExistBearerTokenAuthenticationToken() {
+        this.contextRunner
+            .withUserConfiguration(AzureActiveDirectoryResourceServerClientConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testNotExistOAuth2LoginAuthenticationFilter() {
+        this.contextRunner
+            .withUserConfiguration(AzureActiveDirectoryResourceServerClientConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(OAuth2LoginAuthenticationFilter.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testOAuth2AuthorizedClientRepository() {
+        this.contextRunner
+            .withUserConfiguration(AzureActiveDirectoryResourceServerClientConfiguration.class)
+            .run(context -> {
+                final OAuth2AuthorizedClientRepository aadOboRepo = context.getBean(
+                    AADOAuth2OboAuthorizedClientRepository.class);
+                assertThat(aadOboRepo).isNotNull();
+                assertThat(aadOboRepo).isExactlyInstanceOf(AADOAuth2OboAuthorizedClientRepository.class);
+            });
+    }
+
+    @Test
+    public void testClientRegistrationRepository() {
```

Comment:
We should add more test cases for client registrations. We could run the context with more properties, 

```yaml
xxx:
  graph:
     scopes: xxx
```
to check whether does the graph client exist?

Reply:
Means more resource dependencies to test in the sample?

Reply:
I don't get this
> Means more resource dependencies to test in the sample?

Reply:
I mean the sample should configure more clients to use more scopes scenario, right?

Reply:
No, I'm just talking about the test cases.